sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -192,6 +194,48 @@ private[hive] case class HiveGenericUDF(
 
   override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
     copy(children = newChildren)
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val refTerm = ctx.addReferenceObj("this", this)
+    val childrenEvals = children.map(_.genCode(ctx))
+
+    val setDeferredObjects = childrenEvals.zipWithIndex.map {
+      case (eval, i) =>
+        val deferredObjectAdapterClz = classOf[DeferredObjectAdapter].getCanonicalName
+        s"""
+           |if (${eval.isNull}) {
+           |  (($deferredObjectAdapterClz) $refTerm.deferredObjects()[$i]).set(null);
```

Comment:
The initial value of `func` is null. 

`set(null)`  seem to be a protective operation